### ISSUE TITLE
Two small issues about the paper and code

### ISSUE THREAD

**Issue by marscrazy** [tanyuqian/learning-data-manipulation#2]
Thanks for your great jobs on data augmentation.
I am confused by the code  in `augmentation/classifier.py`   ` def finetune_generator(self, example, aug_probs, finetune_batch_size):`
What's the correlation between the reward (Eq.9 in the paper) and the aforementioned code?

And. what is the purpose of the `magic_module.py`

**Comment by tanyuqian** [comment]
Thanks for carefully reading our code.

Let's start from augmentation_main.py-line94, ```generator.finetune_batch()``` is the function to update ```\phi``` (paper Figure 1).

generator.py-line128: get augmented examples for a training data example (paper Equation 9)

```classifier.finetune_generator()``` includes 
* getting ```\theta'(\phi) = \theta + \nabla_{\theta} L(\theta, \phi)``` (line 136-142 is computing ```grads = \nabla_{\theta} L(\theta, \phi)```).
*  getting gradients of ```\phi``` with dev data (line 150-158). 

generator.py-line 131: update ```\phi``` according collected gradients of ```\phi``` from a batch of examples.

As for the ```magic_module.py```, it is for us to get ```L(\theta'(\phi))``` and gradients of ```\phi```:
* ```grads = \nabla_{\theta} L(\theta, \phi)``` -- classifier.py-line140
* ```\theta'(\phi) = \theta + grads``` -- classifier.py-line148
* ```L(\theta'(\phi))``` -- classifier.py-line154
* ```\nabla_{\phi} L(\theta'(\phi))``` -- classifier.py-line158

Hope these can help you. Please let me know if any part is still confusing you.

**Comment by marscrazy** [comment]
Thanks for your help.
